provisioning-workflows/zOS/NetworkResources/Deprovision_Network_Resource.xml

```diff
@@ -0,0 +1,182 @@
+                            }
+                    }
+                </requestBody>
+                <expectedStatusCode>204</expectedStatusCode>
+            </rest>
+            <setVariable name="IP_ADDRESS_ID" scope="instance"></setVariable>
+        </step>
+        <step name="returnPort" optional="false">
+            <title>Return Port</title>
+            <description>Invoke RESTful API to return port back to pool.</description>
+            <condition>
+                <expression>"1" == "1"</expression>
+                <description>Check that previous step has completed</description>
+                <targetStateSet>
+                    <description>Set state to Skipped if PORT_ID is not set.</description>
+                    <extendStateExpression>
+                        <description>Skipped if PORT_ID is null</description>
+                        <expression> ${instance_PORT_ID} == null</expression>
+                        <targetState>Skipped</targetState>
+                    </extendStateExpression>
+                    <extendStateExpression>
+                        <description>Skipped if PORT_ID is null string</description>
+                        <expression> ${instance_PORT_ID} == ""</expression>
+                        <targetState>Skipped</targetState>
+                    </extendStateExpression>
+                </targetStateSet>
+            </condition>
+            <instructions substitution="false">
+            Execute step to return the HTTP port back to z/OS Communications Server.
+            </instructions>
+            <weight>10</weight>
+            <skills>REST</skills>
+            <autoEnable>true</autoEnable>
+            <canMarkAsFailed>false</canMarkAsFailed>
+            <rest>
+                <httpMethod>POST</httpMethod>
+                <uriPath substitution="true">/zosmf/resource-mgmt/rest/1.0/rdp/network/port/actions/release</uriPath>
+                <requestBody substitution="true">
+
+                    {
+
+                    "template-uuid" : "${_workflow-templateID}",
+                    "template-name" : "${_workflow-templateName}",
+                    "tenant-id" : "${_workflow-tenantID}",
+                    "network-parms" :
+                            {
+                            "port-id" : "${instance-PORT_ID}"
+                            }
+                    }
+
+                </requestBody>
+                <expectedStatusCode>204</expectedStatusCode>
+            </rest>
+            <setVariable name="PORT_ID" scope="instance"></setVariable>
+        </step>
+        <step name="returnAPPLID" optional="false">
+            <title>Return APPLID</title>
+            <description>Invoke RESTful API to return APPLID back to pool.</description>
+            <condition>
+                <expression>"1" == "1"</expression>
+                <description>Check that previous step has completed</description>
```

Comment:
Are we really checking that the previous step has completed here? This looks like its skipping the step depending on the APPLID_ID. Also same statement about condensing expressions

Reply:
I will fix the comment